builtin/providers/google/resource_sql_database.go

```diff
@@ -67,6 +68,7 @@ func resourceSqlDatabaseCreate(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	err = sqladminOperationWait(config, op, "Insert Database")
+	googleMutexKV.Unlock(instanceMutexKey(project, instance_name))
```

Comment:
(I stand corrected on the unlock in the error check block--this Unlock will fire before the error checking, so another Unlock in the error check block below is unnecessary.